ibis/backends/tests/test_vectorized_udf.py

```diff
@@ -73,26 +137,65 @@ def overwrite_struct_analytic(v, w):
     return v - v.mean(), w - w.mean()
 
 
-@analytic(
-    input_type=[dt.double, dt.double],
-    output_type=dt.Struct(['demean', 'demean_weight'], [dt.double, dt.double]),
-)
+# analytic multi-column UDF
 def demean_struct(v, w):
     assert isinstance(v, pd.Series)
     assert isinstance(w, pd.Series)
     return v - v.mean(), w - w.mean()
 
 
-@reduction(
-    input_type=[dt.double, dt.double],
-    output_type=dt.Struct(['mean', 'mean_weight'], [dt.double, dt.double]),
-)
+def create_demean_struct_udf(result_formatter):
+    return analytic(
+        input_type=[dt.double, dt.double],
+        output_type=dt.Struct(
+            ['demean', 'demean_weight'], [dt.double, dt.double]
+        ),
+    )(_format_struct_udf_return_type(demean_struct, result_formatter))
+
+
+demean_struct_udfs = [
```

Comment:
I think DataFrames with matching names are also easy to support?

Reply:
Perhaps also add a test for mismatching names (error out)

Reply:
Based on our implementation I believe we already support this: https://github.com/ibis-project/ibis/blob/master/ibis/util.py#L119

Actually if demean_struct returns a 2-column pd.DataFrame, the test will currently pass whether or not the returned df has matching names because it just sets the names at the end.

Reply:
I see so if there is no matching names, we will just match by position?

Reply:
Correct, based on the current implementation, coerce_to_dataframe just cares if the result DataFrame has the correct number of columns based on output_type.names.